packages/vscode-container-client/src/test/ContainersClientE2E.test.ts

```diff
@@ -297,17 +307,52 @@ describe('(integration) ContainersClientE2E', function () {
                 // Ignore error if the container doesn't exist
             }
 
+            // Try removing the network if it exists so we don't get a name conflict
+            try {
+                await defaultRunner.getCommandRunner()(
+                    client.removeNetworks({ networks: [testContainerNetworkName] })
+                );
+            } catch (error) {
+                // Ignore error if the network doesn't exist
+            }
+
+            // Try removing the volume if it exists so we don't get a name conflict
+            try {
+                await defaultRunner.getCommandRunner()(
+                    client.removeVolumes({ volumes: [testContainerVolumeName] })
+                );
+            } catch (error) {
+                // Ignore error if the volume doesn't exist
+            }
+
+            // Create a network for the container
+            await defaultRunner.getCommandRunner()(
+                client.createNetwork({ name: testContainerNetworkName })
+            );
+
+            // Create a volume for the container
+            await defaultRunner.getCommandRunner()(
+                client.createVolume({ name: testContainerVolumeName })
+            );
+
             // Create a container that will stay running
+            // For fun we'll add a network, a bind mount, a volume, and some ports to it and verify those in both
+            // it('ListContainersCommand') and it('InspectContainersCommand')
             testContainerId = await defaultRunner.getCommandRunner()(
                 client.runContainer({
                     imageRef: imageToTest,
                     detached: true,
                     name: testContainerName,
+                    network: testContainerNetworkName,
+                    mounts: [
+                        { type: 'bind', source: testContainerBindMountSource, destination: '/data1', readOnly: true },
+                        { type: 'volume', source: testContainerVolumeName, destination: '/data2', readOnly: false }
+                    ],
+                    ports: [{ hostPort: 8080, containerPort: 80 }],
+                    exposePorts: [3000], // Uses the `--expose` flag to expose a port without binding it
+                    publishAllPorts: true, // Which will then get bound to a random port on the host, due to this flag

```

Comment:
Importantly, this is currently broken in Docker Desktop 4.42.x, unless you disable IPv6. See https://github.com/microsoft/vscode-containers/issues/155#issuecomment-3013884574.